public/app/percona/shared/core/reducers/index.ts

```diff
@@ -106,8 +106,14 @@ export const updateSettingsAction = createAsyncThunk(
               args.body.email_alerting_settings!.test_email = undefined;
             }
           }
-          const settings = await SettingsService.setSettings(args.body, args.token);
-          await thunkAPI.dispatch(fetchSettingsAction({ usedPassword: password, testEmail }));
+          let settings;
+          try {
+            settings = await SettingsService.setSettings(args.body, args.token);
+            await thunkAPI.dispatch(fetchSettingsAction({ usedPassword: password, testEmail }));
+          } catch (e) {
+            await thunkAPI.dispatch(fetchSettingsAction({ usedPassword: password, testEmail }));
```

Comment:
I think you've missed the `settings = await ...`, but like I've mentioned above is probably not needed :)